lib/Consul/API/Agent.pm

```diff
@@ -77,6 +81,10 @@ sub force_leave {
 sub check_register {
     my ($self, $check, %args) = @_;
     croak 'usage: $agent->check_register($check, [%args])' if grep { !defined } ($check);
+    {
+        local $Carp::CarpInternal{ (__PACKAGE__) } = 1;
```

Comment:
Ah, this is because smart_new croak()s and I don't want the stack to end up at this line, but instead end up at the caller's line.